src/XPurchase/VirtoCommerce.XPurchase/CartAggregateRepository.cs

```diff
@@ -182,6 +182,9 @@ protected virtual async Task<CartAggregate> InnerGetCartAggregateFromCartAsync(S
             {
                 aggregate.CartProducts[cartProduct.Id] = cartProduct;
             }
+
+            await aggregate.RecalculateAsync();
```

Comment:
We are already recalculating cart in `GrabCartAsync`. Could we include aggregate.CartProducts populating there, so we will recalculate it only once?